src/main/java/com/hubspot/jinjava/lib/fn/MacroFunction.java

```diff
@@ -81,6 +82,9 @@ public Object doEvaluate(
           !interpreter.getContext().getEagerTokens().isEmpty()
         )
       ) {
+        interpreter
+          .getContext()
+          .removeEagerTokens(new HashSet<>(interpreter.getContext().getEagerTokens()));
```

Comment:
Does this remove the eager tokens added elsewhere?

Reply:
We will enter a new scope when we evaluate the macro function. When entering a new scope, the eager tokens don't get carried over to the new child so it is a fresh set. Therefore, we are only removing the tokens that were added within this scope. The tokens that are added elsewhere outside of this macro function's scope will not get removed